src/libcore/iter.rs

```diff
@@ -1673,7 +1673,8 @@ impl<A, B> DoubleEndedIterator for Zip<A, B> where
 #[stable(feature = "rust1", since = "1.0.0")]
 #[derive(Clone)]
 pub struct Map<I, F> {
-    iter: I,
+    /// The underlying iterator
+    pub iter: I,
```

Comment:
Yeah for now I think we definitely don't want to make these fields public, if the iterators below need access to these internals they're probably a bit too high level to want a convenience "back to container" method.